ui/components/data-sources/DeleteDataSourceDialog.js

```diff
@@ -25,18 +38,32 @@ const DeleteDataSourceDialog = ({ showModal, dataSource, filter, mutate, onClose
         });
     };
 
-    const primaryContent = (<p>Are you sure you want to delete this data source?</p>);
     const title = `Delete Data Source ${dataSource && dataSource.name}`;
+    const icon = hasResolvers ? <Icon type="pf" name="warning-triangle-o" /> : null;
+    const primaryContent = (
+        <p style={{ fontSize: 18 }}>
+            Are you sure you want to delete this data source?
+        </p>
+    );
+    const secondaryContent = hasResolvers
+        ? <p>All associated resolvers ({dataSource.resolvers.length}) will be deleted</p>
+        : <p>No resolvers will be deleted</p>;
```

Comment:
Why even show a message when nothing is affected?

Reply:
Just to make it explicit. I pictured a user asking himself if something _else_ was going to happen when clicking on delete.